src/HttpClient/PluggableHttpClientFactory.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\HttpClient;
+
+use Http\Client\Common\Plugin as HttpClientPluginInterface;
+use Http\Client\Common\PluginClient;
+use Http\Client\HttpAsyncClient as HttpAsyncClientInterface;
+use Sentry\Options;
+
+/**
+ * This factory can be used to decorate an HTTP client with a list of plugins.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ *
+ * @deprecated since version 2.3, to be removed in 3.0
+ */
+final class PluggableHttpClientFactory implements HttpClientFactoryInterface
```

Comment:
This class seems an overkill to me. IMHO we should just use the `getDefaultPlugins()` method that I suggested as public and add a `$plugins` optional argumento to `HttpClientFactoryInterface::create`

Reply:
It was done like this before, but since in `3.0` I don't want to support customizing the plugins of the HTTP client out-of-the-box (I don't see a real value in supporting this use case) I don't want to tie the `create` method to the list of plugins. This class will be used automatically to wrap the default factory only if the user added a custom plugin to the builder and we can remove it in the next major version without changing anything else besides the builder itself

Reply:
The value is in the fact that at least some of them are necessary to use the Sentry APIs correctly. Wrapping them there would mean that the transport factory would have the responsibility of knowing the default plugins, and instead the client factory should care about the HTTP client only.

Reply:
I see a bit of confusion here, so I will try to explain how things are organized. There are two factories: the transport factory and the HTTP client factory. The first one has been introduced to encapsulate the logic to create the transport and move it out of the builder. The second factory instead is the most important and its purpose is to create a fully configured HTTP client ready to be used by Sentry to send events to the server. To do such thing is adds certain plugins to the client, and extracting such logic from the factory would make it useless as the configured client would not be working out-of-the-box without them. Additional plugins that the users wants to add instead are not part of the client configuration and it's not responsibility of the default factory to add them to the HTTP client

Reply:
Sure, and that's why I'm suggesting a different approach. With your current one, the TransportFactory is just an empty shell without any proper responsibility; I'm suggesting moving the responsibility of attaching the needed plugins there, so the HTTPClientFactory instead will focus on the pure HTTP client and its options, without worrying about what Sentry server needs.

In this way, the end user will be able to intervene onto those two different concerns (plugins and HTTP client) independently, with a simple substitution of one of those two factories.

Reply:
We discussed it at length via chat, I'm ok with the current implementation.